### PR TITLE
feat: Publish events to topic exchange with routing keys (PLA-1466)

### PR DESCRIPTION
**Pull request by Uxio0**
Switch the events producer from a fanout-only setup to a topic exchange so consumers can subscribe to specific event types / chains / safes instead of filtering every message in application code.

Routing key format:
- `"{chainId}.{type}.{address}" `(address lower-cased, "_" placeholder when absent).

The legacy fanout exchange is bound downstream of the topic exchange with routing_key="#" on first publish (idempotent), so existing consumers keep receiving every event during the migration window with no producer-side duplication or loss. Once consumers have migrated their bindings, the legacy exchange can be removed.

### CODE REVIEW

File: safe_transaction_service/events/services/queue_service.py [hunk 1]
```diff
@@ -24,23 +34,78 @@ def __init__(self):
...
+        except Exception as exc:
+            logger.error(
+                "Could not bind legacy fanout exchange to topic exchange: %s",
+                exc,
+                exc_info=True,
+            )
+            raise
+
+    @staticmethod
+    def _build_routing_key(payload: dict[str, Any]) -> str:
```
Comment (falvaradorodriguez) [inline safe_transaction_service/events/services/queue_service.py:72]: IMO, It would be nice to add a few unit tests for `_build_routing_key`, since it’s the core new logic here. Especially around edge cases like missing values  "_" fallbacks and address lowercasing.

**Review by falvaradorodriguez — DISMISSED** [review]
Great contribution!

File: safe_transaction_service/events/services/queue_service.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/events/services/queue_service.py:72]: Added in b1a038d04b292eba7613eaab7c5013a17d09e2d5

**Review by falvaradorodriguez — APPROVED** [review]
LGTM!
